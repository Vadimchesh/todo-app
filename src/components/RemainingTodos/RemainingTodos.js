import React from 'react';

const RemainingTodos = ({ count }) => {
  const suffix = count === 1 ? '' : 's';

  return (
    <label className='labelCount' data-cy='countItemLeft'>
      {count} item{suffix} left
    </label>
  );
};
export default RemainingTodos;
