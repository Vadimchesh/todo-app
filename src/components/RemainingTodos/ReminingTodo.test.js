import RemainingTodos from './RemainingTodos';
import React from 'react';
import renderer from 'react-test-renderer';

// const setUp = (props) => shallow(<RemainingTodos {...props} />);
let wrapper;
beforeEach(() => {
  wrapper = shallow(<RemainingTodos />);
});

describe('Remmaining todo coun:', () => {
  it('should render one label', () => {
    const label = wrapper.find('label');
    expect(label.length).toBe(1);
  });
  it('should render count', () => {
    const label = wrapper.find('label');
    expect(label.text()).toBe(`items left`);
  });

  it('renders correctly', () => {
    const tree = renderer.create(<RemainingTodos />);
    expect(tree).toMatchSnapshot();
  });
});
