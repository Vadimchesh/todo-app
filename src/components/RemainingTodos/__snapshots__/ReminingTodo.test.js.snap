// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Title component should render RemainingTodos component with props 1`] = `
<label
  className="labelCount"
  data-cy="countItemLeft"
>
  1
   item
  s
   left
</label>
`;

exports[`Title component should render RemainingTodos component without props 1`] = `
<label
  className="labelCount"
  data-cy="countItemLeft"
>
   item
  s
   left
</label>
`;
